### DISCUSSION TITLE
Q&A: gradio version

### DISCUSSION THREAD

**Discussion by kalle07** [lllyasviel/Fooocus#3296] [Q&A] [answered]
i have this message during start-up fooocus
IMPORTANT: You are using gradio version 3.41.2, however version 4.29.0 is available, please upgrade.

i typed in CMD
pip install gradio

but same error

**Comment by mashb1t** [comment, marked as answer]
https://github.com/lllyasviel/Fooocus/issues/2911#issuecomment-2108362279
Please ignore and leave it be OR If you have to time to upgrade (basically rework of the whole frontend) go ahead, looking forward to it ^^

**Comment by david-peter-smith** [comment]
It is a coding best practice to upgrade dependencies on a regular basis (i.e., annually) so that security patches and bug fixes in dependencies are migrated to the project. I recommend this be a regular task.

**Reply by mashb1t** [reply]
@david-peter-smith all packages except gradio have been updated to the latest compatible version in v2.5.0. A monthly dependabot automation workflow has already been set up, which can be found in the [actions tab](https://github.com/lllyasviel/Fooocus/actions). Python 3.12 compatibility has been established (see [here](https://github.com/lllyasviel/Fooocus/pull/2927#issuecomment-2231101766)), but used packages use an incompatible and non-standardized versioning scheme, which also does not allow pip 24.1 to be compatible.
Updating Gradio means rewriting the whole frontend, which currently none of the devs has time for (outdated PR [here](https://github.com/lllyasviel/Fooocus/pull/961)). I suggest you or the community support in this matter as you already have peaked interest in the topic. Much appreciated, thanks!
